optimum/onnxruntime/modeling_decoder.py

```diff
@@ -263,20 +288,57 @@ def prepare_io_binding(
 
         return io_binding, output_shapes, output_buffers
 
+    def prepare_inputs_for_merged(
+        self, input_ids: Optional[torch.LongTensor] = None, past_key_values: Optional[List[torch.FloatTensor]] = None
+    ):
+
+        # Prepare use cache
+        if past_key_values is not None and self.use_merged:  # Uses "with past" branch of a merged decoder
+            use_cache = torch.full((1,), True).to(self._device)
+        elif self.use_merged:  # Uses "no past" branch of a merged decoder
+            use_cache = torch.full((1,), False).to(self._device)
+        else:  # Uses separate decoders
+            use_cache = None
+
+        # Prepare past key values
+        is_dummy = False
+        if (
+            self.use_merged and past_key_values is None
+        ):  # Generate dummy past for the first forward if uses a merged decoder
+            batch_size = input_ids.size(0)
+            num_attention_heads = self.normalized_config.num_attention_heads
+            hidden_size = self.normalized_config.hidden_size
+            embed_size_per_head = hidden_size // num_attention_heads
+            shape = (batch_size, num_attention_heads, 1, embed_size_per_head)  # "1" is the dummy sequence length
+            key_or_value = torch.zeros(shape, dtype=torch.float32).to(self._device)
+            past_key_values = [key_or_value for _ in range(len(self.key_value_input_names))]
+            is_dummy = True
+
+        return use_cache, past_key_values, is_dummy
+
     @add_start_docstrings_to_model_forward(DECODER_INPUTS_DOCSTRING)
     def forward(
         self,
         input_ids: torch.LongTensor,
         attention_mask: Optional[torch.LongTensor] = None,
         past_key_values: Optional[Tuple[Tuple[torch.FloatTensor]]] = None,
     ) -> CausalLMOutputWithCrossAttentions:
+
         # Flatten the past_key_values
         if past_key_values is not None:
             past_key_values = [past_key_value for pkv_per_layer in past_key_values for past_key_value in pkv_per_layer]
 
+        # Prepare inputs if uses merged decoder
+        use_cache, past_key_values, is_dummy = self.prepare_inputs_for_merged(input_ids, past_key_values)
```

Comment:
I think it is better to have the `if self.use_merged` here to make things explicit

Reply:
Actually, I think it can be confusing to have `use_cache` both for NoneType, bool, tensor. From what I understand, `use_cache` is an input to the merged without/with past models. I think it would be better to give an other name to avoid giving the same.